src/openassetio-core/include/openassetio/BatchElementError.hpp

```diff
@@ -107,7 +107,21 @@ class BatchElementError final {
      * entity-based operations on a valid @ref entity_reference that
      * fail for some reason.
      */
-    kEntityResolutionError = OPENASSETIO_BatchErrorCode_kEntityResolutionError
+    kEntityResolutionError = OPENASSETIO_BatchErrorCode_kEntityResolutionError,
+
+    /**
+     * Error code response from @ref preflight if the provided
+     * @fqref{TraitsData} "traits data" hint holds insufficient
+     * information.
+     *
+     * This will occur when a workflow requires information up-front, so
+     * that the manager can prepare for upcoming @ref resolve "resolves"
+     * to inform @fqref{Context.Access.kWrite} "write operations" and
+     * eventual @ref register "registration" of the final data, but
+     * insufficient information was given to service those expected
+     * upcoming operations.
```

Comment:
[Consider][Minor] I wonder if spelling out the resolves->write operations->register flow here is more confusing than helpful.

Perhaps just something like?
```
... so that the manager can service upcoming write-workflow operations.
```

Reply:
Shortened in daa8a1ac, relying on Glossary entry link to provide more info if required.